tests/core/test_querydb.py

```diff
@@ -4,63 +4,74 @@
 
 def test_querydb_multiple_instances():
     """Accessing multiple instances simultaneously must work."""
-    cxg = ln.QueryDB("laminlabs/cellxgene")
-    lamindata = ln.QueryDB("laminlabs/lamindata")
-    qs1 = cxg.Artifact.filter(suffix=".h5ad")
-    qs2 = lamindata.Artifact.filter(suffix=".zarr")
+    cxg_db = ln.QueryDB("laminlabs/cellxgene")
+    lamindata_db = ln.QueryDB("laminlabs/lamindata")
+    qs1 = cxg_db.Artifact.filter(suffix=".h5ad")
+    qs2 = lamindata_db.Artifact.filter(suffix=".zarr")
     assert qs1._db != qs2._db
 
 
 def test_querydb_bionty():
     """Querying a record from bionty must work."""
-    cxg = ln.QueryDB("laminlabs/cellxgene")
-    assert len(cxg.Gene.filter(symbol__startswith="TP53")) > 0
+    cxg_db = ln.QueryDB("laminlabs/cellxgene")
+    assert len(cxg_db.bionty.Gene.filter(symbol__startswith="TP53")) > 0
 
 
 def test_querydb_missing_module():
     """Attempting to access an attribute that comes from a missing module must error."""
-    db = ln.QueryDB("laminlabs/lamin-site-assets")  # instance without bionty
+    site_assets_db = ln.QueryDB(
+        "laminlabs/lamin-site-assets"
+    )  # instance without bionty
 
     with pytest.raises(AttributeError) as e:
-        db.Gene.first()
+        site_assets_db.bionty.Gene.first()
 
     assert (
-        "Registry 'Gene' not found in installed modules for instance 'laminlabs/lamin-site-assets'."
+        "Schema 'bionty' not available in instance 'laminlabs/lamin-site-assets'."
         in str(e.value)
     )
 
 
 def test_querydb_instantiate_class():
     """Attempting to instantiate a class must error."""
-    cxg = ln.QueryDB("laminlabs/cellxgene")
+    cxg_db = ln.QueryDB("laminlabs/cellxgene")
     with pytest.raises(TypeError) as e:
-        cxg.Artifact()
+        cxg_db.Artifact()
     assert (
         "Cannot instantiate Artifact from QueryDB. Use Artifact.filter(), Artifact.get(), etc. to query records."
         in str(e.value)
     )
 
 
-def test_querydb_rejects_lowercase():
-    """Accessing registries with lowercase names must fail."""
-    cxg = ln.QueryDB("laminlabs/cellxgene")
+@pytest.mark.parametrize(
+    "attr,expected_msg",
+    [
+        ("artifacts", "Registry 'artifacts' not found"),
+        ("foo", "Registry 'foo' not found"),
+        ("celltype", "Registry 'celltype' not found"),
+    ],
+)
+def test_querydb_rejects_invalid_attributes(attr, expected_msg):
+    """Accessing invalid attributes must fail."""
+    cxg_db = ln.QueryDB("laminlabs/cellxgene")
     with pytest.raises(AttributeError) as e:
-        cxg.artifacts  # noqa: B018
-    assert "Registry names must be capitalized and singular." in str(e.value)
+        getattr(cxg_db, attr)
+    assert expected_msg in str(e.value)
 
 
 def test_querydb_cache():
     """Subsequent accesses must return cached wrapper."""
```

Comment:
Can you add one test that asserts that the queryset caching works as expected?

Say you query `Artifact.filter()` once and then another time; the two results should be different.

I'm pretty sure what returns is a copy of a QuerySet but better double check with a small test.